### PR TITLE
Rename hybridtotcld

### PR DESCRIPTION
**Pull request by cgsandford**
Changed name of the derived total cloud field to reflect its components, as defined by @bjwheltor in the metadata standard.  Derived total cloud is now `low_and_medium_type_cloud_area_fraction`, to distinguish it from the raw model `cloud_area_fraction` field.

Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

**Review by bayliffe — COMMENTED** [review]
One query/request.

File: improver/wxcode/wxcode_decision_tree.py [hunk 1]
```diff
@@ -96,9 +98,7 @@ def wxcode_decision_tree():
             "probability_thresholds": [0.5],
             "threshold_condition": ">=",
             "condition_combination": "",
-            "diagnostic_fields": [
-                "probability_of_texture_of_cloud_area_fraction_above_threshold"
-            ],
+            "diagnostic_fields": [TEXTURE_PROB_ABOVE],
```
[the remark this answers is not in the document]
Comment (cgsandford) [reply improver/wxcode/wxcode_decision_tree.py:101]: Ah tidying up!  That sounds like a plan, I'll look at it now.

File: improver/wxcode/wxcode_decision_tree.py [hunk 2]
```diff
@@ -109,9 +119,7 @@ def wxcode_decision_tree():
             "probability_thresholds": [0.5],
             "threshold_condition": ">=",
             "condition_combination": "",
-            "diagnostic_fields": [
-                "probability_of_lwe_precipitation_rate_above_threshold"
-            ],
+            "diagnostic_fields": [PRECIP_PROB_ABOVE],
```
Comment (bayliffe) [inline improver/wxcode/wxcode_decision_tree.py:122]: Thanks for the tidy up Caroline. Whilst lightning is only used in one place (line 89 above), is it worth defining it with all the other variables in the __init__ to keep things together? I think it probably is.

File: improver/wxcode/wxcode_decision_tree.py [hunk 3]
```diff
@@ -231,10 +212,7 @@ def wxcode_decision_tree():
             "probability_thresholds": [0.5, 0.5],
             "threshold_condition": ">=",
             "condition_combination": "AND",
-            "diagnostic_fields": [
-                "probability_of_lwe_precipitation_rate_above_threshold",
-                (LOW_CLOUD_PROB_ABOVE),
-            ],
+            "diagnostic_fields": [PRECIP_PROB_ABOVE, (LOW_CLOUD_PROB_ABOVE),],
```
Comment (bayliffe) [inline improver/wxcode/wxcode_decision_tree.py:215]: I think we can lose the brackets now we are not mixing strings and variables.

File: improver/wxcode/wxcode_decision_tree_global.py [hunk 4]
```diff
@@ -322,10 +276,7 @@ def wxcode_decision_tree_global():
             "probability_thresholds": [0.5, 0.5],
             "threshold_condition": ">=",
             "condition_combination": "AND",
-            "diagnostic_fields": [
-                "probability_of_lwe_precipitation_rate_above_threshold",
-                (LOW_CLOUD_PROB_ABOVE),
-            ],
+            "diagnostic_fields": [PRECIP_PROB_ABOVE, (LOW_CLOUD_PROB_ABOVE),],
```
Comment (bayliffe) [inline improver/wxcode/wxcode_decision_tree_global.py:279]: Again, surplus brackets now.

**Review by bayliffe — CHANGES_REQUESTED** [review]
A couple more tiny comments.

File: improver/wxcode/wxcode_decision_tree.py (hunk 2 above)
Comment (cgsandford) [reply improver/wxcode/wxcode_decision_tree.py:122]: Oops, sorry missed that one.

File: improver/wxcode/wxcode_decision_tree.py (hunk 3 above)
Comment (cgsandford) [reply improver/wxcode/wxcode_decision_tree.py:215]: I'm a bit scared of editing this as I don't understand what the query interpretation is doing.  Could we have a quick chat so I can tidy up properly?

Comment (bayliffe) [reply improver/wxcode/wxcode_decision_tree.py:213]: The brackets are just really ancient legacy that has been preserved endlessly from when the string name was really long and had to be wrapped onto multiple lines (before black relaxed the line length condition), e.g.

https://github.com/metoppv/improver/blame/44d2df465b4a0181e214670c888249b93e052bc4/lib/improver/wxcode/wxcode_decision_tree.py#L277

No harm will come of removing them.

Comment (cgsandford) [reply improver/wxcode/wxcode_decision_tree.py]: OK thank you.  I've tidied this up now.

File: improver/wxcode/wxcode_decision_tree.py [hunk 5]
```diff
@@ -233,10 +210,7 @@ def wxcode_decision_tree():
             "probability_thresholds": [0.5, 0.5],
             "threshold_condition": ">=",
             "condition_combination": "AND",
-            "diagnostic_fields": [
-                "probability_of_lwe_precipitation_rate_above_threshold",
-                ("probability_of_low_type_cloud_area_fraction_above_threshold"),
-            ],
+            "diagnostic_fields": [PRECIP_PROB_ABOVE, (LOW_CLOUD_PROB_ABOVE)],
```
Comment (bayliffe) [inline improver/wxcode/wxcode_decision_tree.py:213]: (Eric Carmen style): All brackets here, don't wanna have, all brackets here, anymore....

Comment (cgsandford) [reply improver/wxcode/wxcode_decision_tree.py:213]: Ugh, I swear I got them all!  OK I'll go through again...  Sorry

**Review by bayliffe — DISMISSED** [review]

**Review by MoseleyS — CHANGES_REQUESTED** [review]
Code changes look fine and all the tests pass, except for the checksums because the latest version of master and acceptance test data have not been merged into their respective branches. Once the merge is done, I'll be happy to approve this PR.

**Review by MoseleyS — APPROVED** [review]

**Pull request merged**